### ISSUE TITLE
Got an error in the experiments table even when dvc.yaml seems to be correct

### ISSUE THREAD

**Issue by shcheklein** [iterative/vscode-dvc#4241]
<img width="1132" alt="Screenshot 2023-07-08 at 4 54 50 PM" src="https://github.com/iterative/vscode-dvc/assets/3659196/be5c3345-aa49-4feb-b013-58b119065b17">

I could have had a merge conflict in dvc.lock a while ago:

```
[version: 1.0.14, 2023-07-08T23:11:46.912Z, pid: 9232] > /usr/local/bin/python -m dvc data status --granular --unchanged --json - FAILED with code 255 (652ms)
'./dvc.lock' is invalid (possible merge conflicts).



While scanning a simple key, in line 12, column 1

  12 <<<<<<< Updated upstream                                                   



Could not find expected ':', in line 13, column 10

  13 │     md5: acc863c0b1d50f969eba8171c96d0c0c                                
```

But since then, there were multiple successful `dvc status` calls:

```
[version: 1.0.14, 2023-07-08T23:50:51.459Z, pid: 24324] > /usr/local/bin/python -m dvc data status --granular --unchanged --json - COMPLETED (4282ms)
```

**Comment by mattseddon** [comment]
I think that message will be generated as a result of `dvc stage list`. What is the output of that command?

**Comment by shcheklein** [comment]
It's too late, can'd grab it now (instance reloaded). 

I didn't see any errors related to it AFAIR.

In this case btw there was a dvc.yaml, I'm not sure why would we show the button at all.

After reset, renew, etc - I can't reproduce anymore. Might be related to all the updates tbh.

**Comment by mattseddon** [comment]
> In this case btw there was a dvc.yaml, I'm not sure why would we show the button at all.

If the `dvc.yaml` is invalid then `dvc stage list` will return an error. Under those circumstances, we show the button because trying to run an experiment will result in an error. We decided to always show the button because we prioritized onboarding. 

When the `dvc.yaml` becomes valid the button should disappear:

https://github.com/iterative/vscode-dvc/assets/37993418/0ff92e5c-8237-482b-a052-54b542015dba

Given the feedback in #4242, the fact that we can't decide on the copy (#4225) and this issue we should consider removing the button altogether from the table. We could replace it with a Pipeline section on the Setup page.
